.gitmodules

```diff
@@ -4,6 +4,9 @@
 [submodule "rabit"]
 	path = rabit
 	url = https://github.com/dmlc/rabit
+[submodule "nccl"]
+	path = nccl
+	url = https://github.com/h2oai/nccl
```

Comment:
Hi tqchen, Rory can weigh-in here.

There are 3 reasons why this is the best solution:
1) Rory decided this was appropriate because the official nvidia nccl doesn't work on windows.  Someone ported nccl to windows, but they broke linux.  Rory and I fixed that up so that both linux and windows are properly supported.
2) I put in a pull request to nvidia/nccl for these changes, but the changes required would force nccl to be a static lib (as required for xgboost) instead of a shared library.  They probably won't like that.  In addition, they very infrequently take pull requests.
3) nccl 2.0 will be out in several months and may better handle some of the windows issues, but until then Rory and my changes are the only existing nccl that properly handles both linux and windows.